### DISCUSSION TITLE
feat: Image Transformation Support.

### DISCUSSION THREAD

**Discussion by jamaluddinrumi** [sveltia/sveltia-cms#102] [Ideas]
it's really promising when i take a look of how sveltia maximize utilizing github.

but, how about image transformation and then store them to disk? i think we need that feature for supporting dynamic srcset image generation for our img elements.

similar feature is provided by [storyblok image](https://www.storyblok.com/tp/optimize-your-storyblok-images-with-image-engine).

**Comment by kyoshino** [comment]
Thanks for checking out Sveltia CMS, @jamaluddinrumi! I’ve been thinking about this, but honestly it’s becoming less of a priority as some popular frameworks like [Next.js](https://nextjs.org/docs/pages/building-your-application/optimizing/images) and [Hugo](https://gohugo.io/content-management/image-processing/) already provide an image optimization method. One of my clients has been using [Uploadcare](https://uploadcare.com/docs/transformations/image/) for their SvelteKit site, but SvelteKit now also has an experimental [image optimizer](https://kit.svelte.dev/docs/images), so the third-party solution is no longer necessary.

That said, I think a basic resizer/transformer/optimizer in the CMS is still useful, for example to prevent very large images from being uploaded by end users. And I understand that not all frameworks, including Vue I guess, provide such built-in functionality.

I’ll keep this on my to-do list!

See also https://github.com/decaporg/decap-cms/issues/7107, https://github.com/decaporg/decap-cms/issues/2808, https://github.com/decaporg/decap-cms/issues/1322.

**Comment by kyoshino** [comment]
I have added an [image conversion method](https://github.com/sveltia/sveltia-cms/blob/main/src/lib/services/utils/media.js#L101) to generate asset thumbnails. I’ll use it to support simple image transformations. Basically, for websites, we just need WebP quality 85 (or SVG).

**Reply by jamaluddinrumi** [reply]
✨ amazing

would later the result be able to used with [unpic](https://unpic.pics) to make `sveltia` works nicely with [astro](https://astro.build)?

**Reply by kyoshino** [reply]
I have never used Unpic, but if you can use it, I think the built-in image translation is not needed 😆 Still, image optimization is nice to have in the CMS.

**Reply by jamaluddinrumi** [reply]
✨ wow, i'm just realized that this feature ([feat: Image Transformation Support.](https://github.com/sveltia/sveltia-cms/discussions/102#top) #102) will make image `srcset` translation in our frontend is not needed anymore. 

i'm stay following the progress.

**Reply by jptrzy** [reply]
@kyoshino Is it available as an option somewhere?

**Reply by kyoshino** [reply]
I haven’t implemented this yet, but it’s one of the features to be added before the 1.0 release. I’ll keep you posted.

**Reply by jptrzy** [reply]
thx

**Comment by jamaluddinrumi** [comment]
**[out of topic]**

the potential of monetizing `sveltia` is so much promising till it can be used as another paid cms alternative as they (paid cms provider) will raising their services sooner or later.

**Reply by kyoshino** [reply]
Sveltia CMS should always be free 😆 The CMS market is crowded, it’s hard to survive though. But my goal is clear: making a great successor to Netlify CMS.

**Comment by kyoshino** [comment]
Image optimizer has landed in [v0.68.0](https://github.com/sveltia/sveltia-cms/releases/tag/v0.68.0) 😊
